kong/plugins/ai-prompt-guard/handler.lua

```diff
@@ -0,0 +1,42 @@
+local _M = {}
+
+-- imports
+local kong_meta = require "kong.meta"
+local access_handler = require("kong.plugins.ai-prompt-guard.access")
+--
+
+_M.PRIORITY = 771
+_M.VERSION = kong_meta.version
+
+function _M:access(conf)
+  kong.log.debug("IN: ai-prompt-guard/access")
+  kong.service.request.enable_buffering()
+  kong.ctx.shared.prompt_guarded = true
+
+  -- if plugin ordering was altered, receive the "decorated" request
+  local request, err
+  if not kong.ctx.replacement_request then
+    request, err = kong.request.get_body("application/json")
+
+    if err then
+      do_bad_request("ai-prompt-guard only supports application/json requests")
```

Comment:
1. do_bad_request is nil
2. you may want to log an error with these details but i'm not sure if gateway operator wants to let someone know that 'ai-prompt-guard' is in use, that only helps users bypass it (using a little bit of creative)